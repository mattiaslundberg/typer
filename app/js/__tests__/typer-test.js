import React from 'react'
import Typer from '../typer'
import sd from 'skin-deep'

describe('TyperTest', () => {
  const render = (text) => {
    let fullText = text.split("")
    return sd.shallowRender(
      <Typer fullText={fullText} />
    )
  }

  it('Renders initial text', () => {
    let tree = render("hello")

    expect(tree.getRenderOutput()).toMatchSnapshot()
  })

  it('Renders result', () => {
    let component = render("hello")
    let instance = component.getMountedInstance()

    instance.setState({
      startedTyping: 100,
      finished: 300,
    })

    expect(component.getRenderOutput()).toMatchSnapshot()
  })
})
