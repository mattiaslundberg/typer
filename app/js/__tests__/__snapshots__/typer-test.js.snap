exports[`TyperTest Renders initial text 1`] = `
<div
  className="typer">
  <function StatusDisplay(props) {/* istanbul ignore next */_classCallCheck(this, StatusDisplay);return _possibleConstructorReturn(this, (StatusDisplay.__proto__ || Object.getPrototypeOf(StatusDisplay)).call(this,
      props));
    }
    fullText={
      Array [
        "h",
        "e",
        "l",
        "l",
        "o"
      ]
    }
    startTime={undefined}
    typedSequence="" />
  <function Input() {/* istanbul ignore next */_classCallCheck(this, Input);return _possibleConstructorReturn(this, (Input.__proto__ || Object.getPrototypeOf(Input)).call(this));
  
    }
    onType={[Function bound onType]} />
</div>
`;

exports[`TyperTest Renders result 1`] = `
<div
  className="typer">
  <function ResultDisplay() {/* istanbul ignore next */_classCallCheck(this, ResultDisplay);return _possibleConstructorReturn(this, (ResultDisplay.__proto__ || Object.getPrototypeOf(ResultDisplay)).call(this));
  
    }
    textLength={5}
    time={200} />
</div>
`;
